File: lib/leetcode/easy/SummaryRanges.java
package lib.leetcode.easy;

import java.util.ArrayList;
import java.util.List;

class Solution {

    public List<String> summaryRanges(int[] nums) {

        List<String> answers = new ArrayList<String>();
        int i = 0;
        while(i < nums.length) {
            int start = nums[i];
            while (i < nums.length -1 && nums[i]+1==nums[i+1]){
                i++;
            }
            if(start != nums[i]) {
                answers.add(Integer.toString(start)+"->"+Integer.toString(nums[i]));
            }else {
                answers.add(Integer.toString(i));
            }
            i++;
        }
        return answers;
    }


} 

File: lib/leetcode/easy/summary_ranges.dart
List<String> summaryRange(List<int> nums) {
  List<String> answer = [];

  int i = 0;
  while (i < nums.length) {
    final start = nums[i];
    while (i < nums.length && nums[i] + 1 == nums[i + 1]) {
      i++;
    }
    if (start != nums[i]) {
      answer.add('$start->${nums[i]}');
    } else {
      answer.add(start.toString());
    }
    i++;
  }
  return answer;
}
